README.md

```diff
@@ -28,6 +28,8 @@ brew install jq
 
 **Installing and running binaries**
 
+This repository consists of multiple go modules. In order to have a smooth development experience, please add a `go.work` file. It is enough to copy and rename the `go.work.example` file.
```

Comment:
The go.work file is also in this PR, this makes me think it should not. Should we remove the go.work file itself from the PR?

Reply:
Ideally, the `go.work` file should not be commited.

It is commited for now because the testing pipelines fail if you don't commit a go work file. The `go` tool sees that there are 2 modules but does not know how to build and execute normal tests without instructions from `go.work`.

This can also be fixed by some `make` magic, e.g.:
```bash
# create a go.work file on the fly by renaming go.work.example
mv ./go.work.example ./go.work
go test (...)
```

I opted to not do that now since it would be more confusing than just commiting the file directly.

Notice how none of the `make` commands are actually changed, they just point to a different path (`./e2e` instead of `./testing/e2e`).

Reply:
Ah, makes sense. Then maybe let's change that line in the README to account for the fact that the go.work file is in the repo already and keep it like this for now?